fontdrasil/src/orchestration.rs

```diff
@@ -11,7 +11,10 @@ pub const MISSING_DATA: &str = "Missing data, dependency management failed us?";
 
 /// A rule that represents whether access to something is permitted.
 #[derive(Clone)]
-pub enum Access<I> {
+pub enum Access<I>
+where
+    I: Debug,
```

Comment:
I assume something happened, here? There's a major convention in Rust to avoid trait bounds on types wherever possible, and leave them for impl blocks; for instance even `HashMap` does not have a constraint that the key implement `Hash`. Looking through this I assume you ran into some problem that pushed you in this direction, but maybe there's an alternative?

Reply:
I don't need to commit that, will fix. I underappreciated the power of where clauses until the PR. I did run into the traditional desire for a negative clause, but I can see how that might have some sharp edges if added.

> even HashMap does not have a constraint that the key implement Hash

I agree with your point but am inclined to pointlessly dispute this example. The key not having to implement hash is pretty intrinsic to a design where you have a separate hasher instead of keys that that know how to hash themselves.

Reply:
Even with a custom hasher, keys need to implement the `Hash` trait. The [impl block that constains methods like `insert`](http://doc.rust-lang.org/1.70.0/src/std/collections/hash/map.rs.html#732-1193) requires that keys impl `Hash`.